lib/Test/MockFile.pm

```diff
@@ -307,17 +472,56 @@ sub _strict_mode_violation {
     my @stack = _get_stack();
     return unless scalar @stack;    # skip the package
 
+    my $filename;
+
     # check it later so we give priority to authorized_strict_mode_packages
     my $file_arg = file_arg_position_for_command( $command, $at_under_ref );
 
-    my $filename = scalar @$at_under_ref <= $file_arg ? '<not specified>' : $at_under_ref->[$file_arg];
+    if ( $file_arg >= 0 ) {
+        $filename = scalar @$at_under_ref <= $file_arg ? '<not specified>' : $at_under_ref->[$file_arg];
+    }
 
     # Ignore stats on STDIN, STDOUT, STDERR
-    return if $filename =~ m/^\*?(?:main::)?[<*&+>]*STD(?:OUT|IN|ERR)$/;
+    return if defined $filename && $filename =~ m/^\*?(?:main::)?[<*&+>]*STD(?:OUT|IN|ERR)$/;
+
+    my $path = _abs_path_to_file($filename);
+
+    my $context = {
+        command      => $command,
+        filename     => $path,
+        at_under_ref => $at_under_ref
+    };    # object
+
+    my $pass = _validate_strict_rules($context);
+    return if $pass;
+
+    croak("Unknown strict mode violation for $command") if $file_arg == -1;
 
     confess("Use of $command to access unmocked file or directory '$filename' in strict mode at $stack[1] line $stack[2]");
 }
 
+sub _validate_strict_rules {
+    my ($context) = @_;
+
+    # rules dispatch
+    foreach my $rule (@STRICT_RULES) {
+        $context->{'filename'} =~ $rule->{'file_rule'}
```

Comment:
note ` $context->{'filename'}` can be `undef`, please use
`defined  $context->{'filename'} && ..`

Reply:
When will `$context->{'filename'}` be `undef`?

Reply:
when we do not know the filename, which happens for several commands